src/main/java/com/thebois/models/beings/Pawn.java

```diff
@@ -22,19 +28,24 @@
      * @param destination   initial destination to travel to.
      * @param random        the generator of random numbers.
      * @param pathFinder    The generator of paths to positions in the world.
+     * @param finder          Finder used to locate things in the world
      */
     public Pawn(
         final Position startPosition,
         final Position destination,
         final Random random,
-        final IPathFinder pathFinder) {
+        final IPathFinder pathFinder,
+        final IFinder finder) {
         super(startPosition, destination, pathFinder);
         this.random = random;
+        setFinder(finder);
     }
 
     @Override
     public void update() {
         super.update();
+        updateClosestStructure();
```

Comment:
style

You're not really updating the structure so much as you are *finding* the closest structure, right?